File: frontend/src/main/java/com/innertrack/model/TherapistProfile.java
package com.innertrack.model;

import java.time.LocalDateTime;

public class TherapistProfile {
    private int id;
    private int userId;
    private String specialization;
    private String licenseNumber;
    private String bio;
    private String address;
    private Double latitude;
    private Double longitude;
    private String phone;
    private Integer sessionRate;
    private String availableDays;   // comma-separated: MON,TUE,WED
    private LocalDateTime createdAt;

    public TherapistProfile() {}

    public TherapistProfile(int userId) {
        this.userId = userId;
    }

    // ── Getters & Setters ─────────────────────────────────────

    public int getId()                           { return id; }
    public void setId(int id)                    { this.id = id; }

    public int getUserId()                       { return userId; }
    public void setUserId(int userId)            { this.userId = userId; }

    public String getSpecialization()            { return specialization; }
    public void setSpecialization(String v)      { this.specialization = v; }

    public String getLicenseNumber()             { return licenseNumber; }
    public void setLicenseNumber(String v)       { this.licenseNumber = v; }

    public String getBio()                       { return bio; }
    public void setBio(String bio)               { this.bio = bio; }

    public String getAddress()                   { return address; }
    public void setAddress(String address)       { this.address = address; }

    public Double getLatitude()                  { return latitude; }
    public void setLatitude(Double latitude)     { this.latitude = latitude; }

    public Double getLongitude()                 { return longitude; }
    public void setLongitude(Double longitude)   { this.longitude = longitude; }

    public String getPhone()                     { return phone; }
    public void setPhone(String phone)           { this.phone = phone; }

    public Integer getSessionRate()              { return sessionRate; }
    public void setSessionRate(Integer rate)     { this.sessionRate = rate; }

    public String getAvailableDays()             { return availableDays; }
    public void setAvailableDays(String days)    { this.availableDays = days; }

    public LocalDateTime getCreatedAt()          { return createdAt; }
    public void setCreatedAt(LocalDateTime v)    { this.createdAt = v; }

    public boolean hasLocation() {
        return latitude != null && longitude != null;
    }

    /** Human-readable availability string for display in client map panel */
    public String getFormattedDays() {
        if (availableDays == null || availableDays.isBlank()) return "—";
        return availableDays
                .replace("MON", "Lun").replace("TUE", "Mar").replace("WED", "Mer")
                .replace("THU", "Jeu").replace("FRI", "Ven").replace("SAT", "Sam")
                .replace("SUN", "Dim").replace(",", " · ");
    }
}

File: backend/src/Entity/TherapistProfile.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'therapist_profile')]
class TherapistProfile
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\OneToOne(targetEntity: User::class, inversedBy: 'therapistProfile')]
    #[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'id', nullable: false)]
    private User $user;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $specialization = null;

    #[ORM\Column(name: 'license_number', length: 100, nullable: true)]
    private ?string $licenseNumber = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $bio = null;

    #[ORM\Column(name: 'created_at', type: 'datetime')]
    private \DateTimeInterface $createdAt;

    #[ORM\Column(length: 500, nullable: true)]
    private ?string $address = null;

    #[ORM\Column(type: 'float', nullable: true)]
    private ?float $latitude = null;

    #[ORM\Column(type: 'float', nullable: true)]
    private ?float $longitude = null;

    #[ORM\Column(length: 20, nullable: true)]
    private ?string $phone = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
    }

    public function getId(): ?int { return $this->id; }
    public function getUser(): User { return $this->user; }
    public function setUser(User $user): self { $this->user = $user; return $this; }
    public function getSpecialization(): ?string { return $this->specialization; }
    public function setSpecialization(?string $specialization): self { $this->specialization = $specialization; return $this; }
    public function getLicenseNumber(): ?string { return $this->licenseNumber; }
    public function setLicenseNumber(?string $licenseNumber): self { $this->licenseNumber = $licenseNumber; return $this; }
    public function getBio(): ?string { return $this->bio; }
    public function setBio(?string $bio): self { $this->bio = $bio; return $this; }
    public function getCreatedAt(): \DateTimeInterface { return $this->createdAt; }
    public function getAddress(): ?string { return $this->address; }
    public function setAddress(?string $address): self { $this->address = $address; return $this; }
    public function getLatitude(): ?float { return $this->latitude; }
    public function setLatitude(?float $latitude): self { $this->latitude = $latitude; return $this; }
    public function getLongitude(): ?float { return $this->longitude; }
    public function setLongitude(?float $longitude): self { $this->longitude = $longitude; return $this; }
    public function getPhone(): ?string { return $this->phone; }
    public function setPhone(?string $phone): self { $this->phone = $phone; return $this; }

    public function hasLocation(): bool
    {
        return $this->latitude !== null && $this->longitude !== null;
    }
}
